scripts/contributor_dashboard_debug_test.py

```diff
@@ -0,0 +1,260 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/contributor_dashboard_debug.py."""
+
+from __future__ import annotations
+
+from core import feconf
+from core.controllers import base
+from core.domain import opportunity_services
+from core.domain import question_fetchers
+from core.domain import skill_services
+from core.domain import story_fetchers
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.domain import user_services
+from core.platform import models
+from core.platform.auth import firebase_auth_services_test
+from core.tests import test_utils
+
+from scripts import contributor_dashboard_debug
+
+import firebase_admin
+from firebase_admin import auth as firebase_auth
+
+from typing import Any, Dict, List
+
+auth_services = models.Registry.import_auth_services()
+
+
+class ContributorDashboardDebugInitializerTests(test_utils.GenericTestBase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.firebase_sdk_stub = (
+            firebase_auth_services_test.FirebaseAdminSdkStub())
+        self.firebase_sdk_stub.install(self)
+        self.contributor_dashboard_debug = (
+            contributor_dashboard_debug.ContributorDashboardDebugInitializer(
+                base_url=''))
+        # Functions signup(), add_user_role(), set_curriculum_admins() must be
+        # called before login(), because those functions will call logout().
+        self.tested_email = 'testuser@example.com'
+        self.tested_username = 'testuser'
+        self.signup(self.tested_email, self.tested_username)
+        self.add_user_role(
+            self.SUPER_ADMIN_USERNAME, feconf.ROLE_ID_QUESTION_ADMIN)
+        self.set_curriculum_admins([self.SUPER_ADMIN_USERNAME]) # type: ignore
+        self.login(self.SUPER_ADMIN_EMAIL, is_super_admin=True)
+        self.contributor_dashboard_debug.csrf_token = self.get_new_csrf_token() # type: ignore
+
+        self.request_swap = self.swap(
+            self.contributor_dashboard_debug.session,
+            'request',
+            self.mock_request)
+
+    def tearDown(self) -> None:
+        self.firebase_sdk_stub.uninstall()
+        self.logout()
+        super().tearDown()
+
+    def mock_request(
+        self,
+        method: str,
+        url: str,
+        params: Dict[str, Any] | None = None,
+        headers: Dict[str, Any] | None = None
+        ) -> Any:
```

Comment:
```suggestion
    ) -> Any:
```

Fix indentation to make it clear where the function signature ends. Please fix indentation elsewhere in this file too

Reply:
Thanks, I fixed them.